### PR TITLE
Added autoscaling group

### PR DESCRIPTION
**Pull request by goruha**
## what
* Add the ability to create capacity providers from external ec2 autoscale groups

## why
* To have EC2 capacity providers with scaling not managed by ECS cluster

### CODE REVIEW

File: examples/complete/main.tf [hunk 1]
```diff
@@ -38,3 +38,31 @@ module "ecs_cluster" {
     }
   }
 }
+
+data "aws_ssm_parameter" "ami" {
+  name = "/aws/service/ecs/optimized-ami/amazon-linux-2/recommended/image_id"
```
Comment (nitrocode) [inline examples/complete/main.tf:43]: Add count enabled logic here too please

File: examples/complete/main.tf [hunk 2]
```diff
@@ -38,3 +38,31 @@ module "ecs_cluster" {
...
+}
+
+module "autoscale_group" {
+  source  = "cloudposse/ec2-autoscale-group/aws"
+  version = "0.31.1"
+
+  name    = "external-capacity-provider"
```
Comment (nitrocode) [inline examples/complete/main.tf:50]: Why override name here instead of using context?

File: examples/complete/main.tf [hunk 3]
```diff
@@ -38,3 +38,31 @@ module "ecs_cluster" {
     }
   }
 }
+
+data "aws_ssm_parameter" "ami" {
```
Comment (nitrocode) [inline examples/complete/main.tf:42]: What if we want to override this?

File: examples/complete/main.tf [hunk 4]
```diff
@@ -38,3 +38,31 @@ module "ecs_cluster" {
...
+}
+
+module "autoscale_group" {
+  source  = "cloudposse/ec2-autoscale-group/aws"
+  version = "0.31.1"
+
+  name    = "external-capacity-provider"
+  context = module.this.context
+
+  image_id                    = data.aws_ssm_parameter.ami.value
```
Comment (nitrocode) [inline examples/complete/main.tf:53]: What if we want to override this?

File: examples/complete/main.tf [hunk 5]
```diff
@@ -38,3 +38,31 @@ module "ecs_cluster" {
...
+  min_size                    = 0
+  max_size                    = 2
+  wait_for_capacity_timeout   = "5m"
+  associate_public_ip_address = false
+  user_data_base64            = ""
+
+  # Auto-scaling policies and CloudWatch metric alarms
+  autoscaling_policies_enabled           = true
+  cpu_utilization_high_threshold_percent = "70"
+  cpu_utilization_low_threshold_percent  = "20"
```
Comment (nitrocode) [inline examples/complete/main.tf:67]: We should expose many of these inputs, perhaps in a single map like `asg.min_size`, `asg.max_size`, etc ?

File: examples/complete/main.tf (hunk 3 above)
Comment (goruha) [reply examples/complete/main.tf:42]: This is test - we do not care

File: examples/complete/main.tf (hunk 4 above)
Comment (goruha) [reply examples/complete/main.tf:53]: this is test - we do not care

File: examples/complete/main.tf [hunk 6]
```diff
@@ -38,3 +38,31 @@ module "ecs_cluster" {
     }
   }
 }
+
```
Comment (nitrocode) [inline examples/complete/main.tf:41]: Oh nvm, I assumed this wasnt an example. I thought we were baking the asg directly in this. I like this approach better to keep the ecs cluster separate from the asg.

**Review by nitrocode — APPROVED** [review]

**Pull request merged**